src/core/thread/meshcop_tlvs.hpp

```diff
@@ -55,44 +55,20 @@ class Tlv
 {
 public:
     /**
-     * MeshCoP TLV Types.
-     *
-     */
-    enum Type
-    {
-        kChannel           = 0,   ///< Channel TLV
```

Comment:
We can minimize the diffs here by keeping this enum definition and defining to the new `OT_MESHCOP_*` values.  For example `kChannel = OT_MESHCOP_TLV_CHANNEL`.